api/store.js

```diff
@@ -0,0 +1,176 @@
+        } else {
+          logger.silly(`blob [${key}] does ${exists ? '' : 'not '}exist`);
+          resolve(exists);
+        }
+      });
+    });
+  }
+
+  /**
+   * @async
+   * Gets the data from a specified blob.
+   * @param {String} key - Blob identifier
+   * @returns {Buffer} A buffer of the data in the blob.
+   */
+  async get(key) {
+    return new Promise((resolve, reject) => {
+      logger.silly(`getting data from blob [${key}]`);
+      // TODO - make this a buffer or byte array, but not a string
+      let out = '';
+      const readStream = this.store.createReadStream(key);
+      readStream.on('data', chunk => {
+        out += chunk;
+      });
+      readStream.on('end', () => {
+        resolve(out);
+      });
+      readStream.on('error', e => {
+        logger.error(e);
+        reject(e);
+      });
+    });
+  }
+
+  /**
+   * Get a readable stream for a specified blob.
+   * @param {String} key - Blob identifier
+   * @returns {ReadableStream} A readable stream
+   */
+  getReadStream(key) {
+    logger.silly(`getting a read stream for blob [${key}]`);
+    return this.store.createReadStream(key);
+  }
+
+  /**
+   * @async
+   * Remove a blob
+   * @param {String} key - Blob identifier
+   */
+  async remove(key) {
+    return new Promise(resolve => {
+      logger.silly(`removing blob [${key}]`);
+      this.store.remove(key, () => resolve());
+    });
+  }
+
+  /**
+   * @async
+   * Write data into a blob with a specified key.  The blob
+   * will be created if it does not already exist.
+   * @param {String} key - Blob identifier
+   * @param {(ReadableStream|*)} - The data to write.  If a ReadableStream is
+   *    provided, its contents will be piped into the blob.  For everything
+   *    else, it will be written directly to the blob.
+   * @param {*} opts - Returned value when the write is finished.
+   */
+  async write(key, data, opts) {
+    return new Promise((resolve, reject) => {
+      logger.silly(`writing to blob [${key}]`);
+      const writeStream = this.store.createWriteStream(key);
+      let rejected = false;
+
+      const done = () => {
+        logger.silly('finished writing blob');
+        if (!rejected) {
+          resolve(opts);
+        }
+      };
+
+      const errHandler = e => {
+        rejected = true;
+        logger.error(e);
+        reject(e);
+      };
+
+      if (isReadableStream(data)) {
+        logger.silly('source is a stream');
+        data.pipe(writeStream);
+
+        data.on('close', () => {
+          writeStream.end();
+          done();
+        });
+
+        data.on('error', errHandler);
+        writeStream.on('error', errHandler);
+      } else {
+        logger.silly('source is memory data');
+        writeStream.write(data);
+        writeStream.end(done);
+        writeStream.on('error', errHandler);
+      }
+    });
+  }
+}
+
+const def = new Store(
+  getBlobStore(process.env.STORE_TYPE || '', process.env.STORE_PATH || '')
+);
```

Comment:
Made this a class to make testing easier - can inject a mock blob-store!  But also, created a default instance based on config.